pallets/stream-payment/src/mock.rs

```diff
@@ -0,0 +1,422 @@
+    type SS58Prefix = ();
+    type OnSetCode = ();
+    type MaxConsumers = ConstU32<16>;
+}
+
+parameter_types! {
+    pub ExistentialDeposit: u128 = 1;
+}
+
+impl pallet_balances::Config for Runtime {
+    type MaxReserves = ();
+    type ReserveIdentifier = [u8; 4];
+    type MaxLocks = ();
+    type Balance = Balance;
+    type RuntimeEvent = RuntimeEvent;
+    type DustRemoval = ();
+    type ExistentialDeposit = ExistentialDeposit;
+    type AccountStore = System;
+    type FreezeIdentifier = RuntimeFreezeReason;
+    type MaxFreezes = ConstU32<1>;
+    type RuntimeHoldReason = RuntimeHoldReason;
+    type RuntimeFreezeReason = RuntimeFreezeReason;
+    type MaxHolds = ConstU32<5>;
+    type WeightInfo = ();
+}
+
+#[derive(RuntimeDebug, PartialEq, Eq, Encode, Decode, Clone, TypeInfo, MaxEncodedLen)]
+pub enum StreamPaymentAssetId {
+    Native,
+    Dummy,
+}
+
+pub struct StreamPaymentAssets;
+impl pallet_stream_payment::Assets<AccountId, StreamPaymentAssetId, Balance>
+    for StreamPaymentAssets
+{
+    fn transfer_deposit(
+        asset_id: StreamPaymentAssetId,
+        from: &AccountId,
+        to: &AccountId,
+        amount: Balance,
+    ) -> frame_support::pallet_prelude::DispatchResult {
+        Self::decrease_deposit(asset_id.clone(), from, amount)?;
+        match asset_id {
+            StreamPaymentAssetId::Native => {
+                Balances::transfer(from, to, amount, Preservation::Preserve).map(|_| ())
+            }
+            StreamPaymentAssetId::Dummy => Ok(()),
+        }
+    }
+
+    fn increase_deposit(
+        asset_id: StreamPaymentAssetId,
+        account: &AccountId,
+        amount: Balance,
+    ) -> frame_support::pallet_prelude::DispatchResult {
+        match asset_id {
+            StreamPaymentAssetId::Native => Balances::increase_frozen(
+                &pallet_stream_payment::FreezeReason::StreamPayment.into(),
+                account,
+                amount,
+            ),
+            StreamPaymentAssetId::Dummy => Ok(()),
+        }
+    }
+
+    fn decrease_deposit(
+        asset_id: StreamPaymentAssetId,
+        account: &AccountId,
+        amount: Balance,
+    ) -> frame_support::pallet_prelude::DispatchResult {
+        match asset_id {
+            StreamPaymentAssetId::Native => Balances::decrease_frozen(
+                &pallet_stream_payment::FreezeReason::StreamPayment.into(),
+                account,
+                amount,
+            ),
+            StreamPaymentAssetId::Dummy => Ok(()),
+        }
+    }
+}
+
+#[derive(RuntimeDebug, PartialEq, Eq, Encode, Decode, Clone, TypeInfo, MaxEncodedLen)]
+pub enum TimeUnit {
+    BlockNumber,
+    Timestamp,
+    Never,
+}
+
+pub struct TimeProvider;
+impl pallet_stream_payment::TimeProvider<TimeUnit, Balance> for TimeProvider {
+    fn now(unit: &TimeUnit) -> Option<Balance> {
+        match unit {
+            &TimeUnit::BlockNumber => Some(System::block_number().into()),
+            &TimeUnit::Timestamp => todo!(),
```

Comment:
This is missing tests. I guess setting the timestamp to `block_number * 12` will be good enough.